crates/store/src/db/sql/plan_explainer.rs

```diff
@@ -0,0 +1,161 @@
+use rusqlite::{CachedStatement, Params, Row, Rows, Statement};
+use termtree::Tree;
+use tracing::info;
+
+use crate::db::{connection::Connection, transaction::Transaction};
+
+impl Connection {
+    #[inline]
+    pub fn prepare_cached(&self, sql: &str) -> rusqlite::Result<CachedStatementWithQueryPlan<'_>> {
+        let statement = self.inner().prepare_cached(sql)?;
+
+        Ok(CachedStatementWithQueryPlan {
+            executor: Executor::Connection(self.inner()),
+            sql: sql.into(),
+            statement,
+        })
+    }
+
+    #[inline]
+    pub fn execute<P: Params + Clone>(&self, sql: &str, params: P) -> rusqlite::Result<usize> {
+        self.prepare_cached(sql)?.execute(params)
+    }
+
+    #[inline]
+    pub fn query_row<T, P, F>(&self, sql: &str, params: P, f: F) -> rusqlite::Result<T>
+    where
+        P: Params + Clone,
+        F: FnOnce(&Row<'_>) -> rusqlite::Result<T>,
+    {
+        self.prepare_cached(sql)?.query_row(params, f)
+    }
+}
+
+impl Transaction<'_> {
+    pub fn prepare_cached(&self, sql: &str) -> rusqlite::Result<CachedStatementWithQueryPlan> {
+        let statement = self.inner().prepare_cached(sql)?;
+
+        Ok(CachedStatementWithQueryPlan {
+            executor: Executor::Transaction(self.inner()),
+            sql: sql.into(),
+            statement,
+        })
+    }
+}
+
+enum Executor<'conn> {
+    Connection(&'conn rusqlite::Connection),
+    Transaction(&'conn rusqlite::Transaction<'conn>),
+}
+
+pub struct CachedStatementWithQueryPlan<'conn> {
+    executor: Executor<'conn>,
+    sql: Box<str>,
+    statement: CachedStatement<'conn>,
+}
```

Comment:
We should be able to get rid of this if we only support `Transaction` right?

I'm thinking it just becomes `Transaction::check_query_plan(&self, sql: &str)` which is selectively enabled by `#[cfg(test)]`.